language/move-lang/src/lib.rs

```diff
@@ -577,100 +594,89 @@ impl PassResult {
     pub fn equivalent_pass(&self) -> Pass {
         match self {
             PassResult::Parser(_, _) => Pass::Parser,
-            PassResult::Expansion(_, _) => Pass::Expansion,
-            PassResult::Naming(_, _) => Pass::Naming,
+            PassResult::Expansion(_) => Pass::Expansion,
+            PassResult::Naming(_) => Pass::Naming,
             PassResult::Typing(_) => Pass::Typing,
-            PassResult::HLIR(_, _) => Pass::HLIR,
+            PassResult::HLIR(_) => Pass::HLIR,
             PassResult::CFGIR(_) => Pass::CFGIR,
             PassResult::Compilation(_) => Pass::Compilation,
         }
     }
-
-    pub fn check_for_errors(self) -> Result<Self, Errors> {
-        Ok(match self {
-            result @ PassResult::Parser(_, _)
-            | result @ PassResult::Typing(_)
-            | result @ PassResult::CFGIR(_)
-            | result @ PassResult::Compilation(_) => result,
-            PassResult::Expansion(eprog, errors) => {
-                check_errors(errors)?;
-                PassResult::Expansion(eprog, Errors::new())
-            }
-            PassResult::Naming(nprog, errors) => {
-                check_errors(errors)?;
-                PassResult::Naming(nprog, Errors::new())
-            }
-            PassResult::HLIR(hprog, errors) => {
-                check_errors(errors)?;
-                PassResult::HLIR(hprog, Errors::new())
-            }
-        })
-    }
 }
 
 fn run(
+    compilation_env: &mut CompilationEnv,
     pre_compiled_lib: Option<&FullyCompiledProgram>,
```

Comment:
Maybe we could move this into the CompilationEnv? Thoughts?

Reply:
IMO it feels better here and not in `CompilationEnv` (if we want to keep that is small as possible). Compilation taking the compilation env in seems more in keeping with the rest of the design in the compiler. But I  don't have particularly strong feelings about it, so if you think it should go there I can move it in.

Reply:
I eventually want a big struct called `Compiler` to handle the mess of entry points we have right now. Once that happens it might make sense to move this, but until then, it is probably fine